qcodes/data/data_set.py

```diff
@@ -1,14 +1,17 @@
 """DataSet class and factory functions."""
+from __future__ import annotations
 
 import logging
 import time
 from collections import OrderedDict
 from copy import deepcopy
 from traceback import format_exc
-from typing import Any, Callable, Dict, List, Optional
+from typing import TYPE_CHECKING, Any, Callable, Dict, Optional
```

Comment:
```suggestion
from typing import TYPE_CHECKING, Any, Callable, Dict
```

Reply:
Looks like `from typing import TYPE_CHECKING, Sequence` would be enough